Move response/empty to http/request-structs.rkt.

jessealama opened the pull request:
The documentation incorrectly implies that `response/empty`
is in the `web-server/http/response-structs` module.  You
can see this if you look at

  https://docs.racket-lang.org/search/index.html?q=response%2Fempty

for 7.6. But this isn't true: if you use `response/empty` after `require`ing `web-server/http/response-structs`, you'll get an unbound identifier error. `response/empty` is actually provided from `web-server/http/empty`.

Rather than viewing this as a problem with the
documentation, the solution here is to incorporate
`response/empty`, which is currently defined in a little
module `empty.rkt`, into `response-structs.rkt`. It thus
sites alongside other utilities such as `response/full` and
`response/output`.

jessealama commented:
Pull request for the TR web-server is [here](https://github.com/racket/typed-racket/pull/922).

jessealama commented:
The solution here is just a proposal. I can imagine at least one alternative: provide `response/empty` from `web-server/http`. (A minor change to the docs would be needed.)

jessealama commented:
The build dies because TR is trying to use a file that gets deleted here. This change and the associated TR change depend on each other.

samth commented:
The re-provide solution seems simpler to me.

jeapostrophe commented:
I think the original intention was to provide it from `web-server/http` or correct the docs so that there's no backwards compatibility problems.

jessealama commented:
> I think the original intention was to provide it from `web-server/http` or correct the docs so that there's no backwards compatibility problems.

Sounds good. I've adjusted the docs and fixed the problem by (re)providing `response/empty` from `web-server/http`. I've squashed all the intermediate work into a single commit for the sake of cleanliness.

Backwards compatibility is preserved. `response/empty` never was really available in `web-server/http/response-structs`, despite what the docs implied, but it was available if you `require`d the undocumented `web-server/http/empty` module. That is preserved here.

jeapostrophe commented:
Thanks... this is a bit messy but I think this might be the right thing.
